packages/fetchai/protocols/fipa/message.py

```diff
@@ -111,6 +111,12 @@ def proposal(self) -> List[Description]:
         assert self.is_set("proposal"), "proposal is not set."
         return cast(List[Description], self.get("proposal"))
 
+    @property
```

Comment:
Please remove the dependency on tx_nonce and vice versa here. `tx_nonce` should be a property of the proposal.